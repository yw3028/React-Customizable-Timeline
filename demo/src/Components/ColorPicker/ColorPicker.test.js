import { cleanup, render, screen } from '@testing-library/react';
import React from 'react';

import ColorPicker from './ColorPicker';

const renderComponent = (colors, setColors) =>
  render(<ColorPicker colors={colors} setColors={setColors} />);

afterEach(() => {
  cleanup();
});

describe('ColorPicker component', () => {
  it('renders correctly', () => {
    const component = renderComponent(defaultColors);
    expect(component.baseElement).toMatchSnapshot();
  });
});

const defaultColors = {
  primaryColor: '#625261',
  secondaryColor: '#F5F5DC',
  backgroundColor: '#FFFFFF',
};
