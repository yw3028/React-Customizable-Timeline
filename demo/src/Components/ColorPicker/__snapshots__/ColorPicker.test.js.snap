// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker component renders correctly 1`] = `
<body>
  <div>
    <div
      class="color_section"
    >
      <div
        class="color_section_colors"
      >
        <h4>
           
          Primary
           
        </h4>
        <div
          style="display: flex;"
        >
          <input
            name="primaryColor"
            type="color"
            value="#625261"
          />
        </div>
        <input
          class="hex-code"
          name="primaryColor"
          placeholder="#625261"
          type="text"
          value="#625261"
        />
      </div>
      <div
        class="color_section_colors"
      >
        <h4>
           
          Secondary
           
        </h4>
        <div
          style="display: flex;"
        >
          <input
            name="secondaryColor"
            type="color"
            value="#F5F5DC"
          />
        </div>
        <input
          class="hex-code"
          name="secondaryColor"
          placeholder="#F5F5DC"
          type="text"
          value="#F5F5DC"
        />
      </div>
      <div
        class="color_section_colors"
      >
        <h4>
           
          Background
           
        </h4>
        <div
          style="display: flex;"
        >
          <input
            name="backgroundColor"
            type="color"
            value="#FFFFFF"
          />
          <button
            class="button-transparent"
            name="backgroundColor"
          />
        </div>
        <input
          class="hex-code"
          name="backgroundColor"
          placeholder="#FFFFFF"
          type="text"
          value="#FFFFFF"
        />
      </div>
    </div>
  </div>
</body>
`;
